### ISSUE TITLE
Inconsistent Angle Mode in WEBGL for rotateX, rotateY, rotateZ

### ISSUE THREAD

**Issue by PixelBaer** [processing/p5.js#2422]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [x] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: p5.js v0.5.16 October 11, 2017
- Web browser and version: Chrome 63.0.3239.84
- Operating System: Windows 10 Pro
- Steps to reproduce this:

The documentation says rotateX/Y/Z accepts only radiants as arguments. But if you change angleMode to DEGREES it seems to work with the Y and Z axis but not with the X axis.

Here's a simple demo with sliders for X/Y/Z rotation. Notice how  y and z works differently than x.

```javascript
var div, divX, divY, divZ, sliderX, sliderY, sliderZ;

function setup() {
  createCanvas(600, 400, WEBGL);
  background(51);
  angleMode(DEGREES);
  
  div = createDiv('Sliders:');
  divX = createDiv('X: ');
  divY = createDiv('Y: ');
  divZ = createDiv('Z: ');
  divX.parent(div);
  divY.parent(div);
  divZ.parent(div);
  sliderX = createSlider(0, 180, 30, 0.1);
  sliderX.parent(div);
  sliderY = createSlider(0, 360, 45, 0.1);
  sliderY.parent(div);
  sliderZ = createSlider(0, 360, 0, 0.1);
  sliderZ.parent(div);
}

function draw() {
  background(51);
  
  divX.html('X: '+sliderX.value());
  divY.html('Y: '+sliderY.value());
  divZ.html('Z: '+sliderZ.value());
  
  rotateX(sliderX.value());
  rotateY(sliderY.value());
  rotateZ(sliderZ.value());
  
  noStroke();
  normalMaterial();
  box(50, 100, 120); 

}
```

**Comment by mlarghydracept** [comment]
Hm strange behavior. The model view matrix seems to be spastically swinging between values with those sliders. I managed to fix it in your code by setting the start value of each slider to 0 and the step size to 0.01. Doesn't explain underlying behavior between the slider and the model view matrix but will work for you for now.

**Comment by Spongman** [comment]
It's because the mayrix math functions are generating matrices that no longer contain a reference to the p5 instance and therefore the angleMode is forgotten.

**Comment by mlarghydracept** [comment]
fixed with #2439 !
